lib/apipie/swagger_generator.rb

```diff
@@ -504,7 +507,8 @@ def save_field(entry, openapi_key, v, apipie_key=openapi_key, translate=false)
       end
 
       if !in_schema
-        swagger_def[:in] = param_desc.options.fetch(:in, @default_value_for_param_in)
+        # the "name" and "in" keys can only be set on root parameters (non-nested)
+        swagger_def[:in] = @default_value_for_param_in if name.present?
```

Comment:
I'm not sure I totally follow the dependency on name here

Reply:
root parameters [must have a name and in key](https://swagger.io/docs/specification/2-0/describing-parameters/) where nested parameters (ex data.attributes.foo) will not